gcc/testsuite/rust/compile/torture/extern_mod4.rs

```diff
@@ -0,0 +1,17 @@
+// { dg-additional-options "-w" }
```

Comment:
I suggest to avoid `-w` and instead spell out the current set of diagnostics: while `-w` does get rid of those you're not interested in at this time, it also forever prevents noticing those that may bogusly appear in the future.

Reply:
Oh, is the problem here (and likewise in the other test cases) that we're getting emitted diagnostics not for this `*.rs` file but for the module:

```
$ build-gcc/gcc/xgcc -Bbuild-gcc/gcc/ source-gcc/gcc/testsuite/rust/compile/torture/extern_mod4.rs 
source-gcc/gcc/testsuite/rust/compile/torture/modules/mod.rs:1:5: warning: unused name 'return_12'
    1 | pub fn return_12() -> i32 {
      |     ^
```

Hmm.  Will have to think about how to deal with that...

Reply:
Though, this diagnostic is bogus here, so should fix the compiler to not emit it, right?  (But will need a way to deal with such cross-file diagnostics, generally.)

Reply:
The problem is indeed that we get this extra warning on the external file

One of the issues is that right now, Ribs are getting duplicated in modules. This is what @philberty was explaining to me, and I'm paraphrasing, but what you get is the following structure of names/Ribs:
```
[modules::return_12]
[return_12]
```
So we actually have two Ribs pointing to the same function. And since the test uses `modules::return_12()`, well one of them is correctly marked as unused :D